File: app/src/main/java/cn/com/woong/readhub/domain/RxSchedulers.java
package cn.com.woong.readhub.domain;


import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.annotations.NonNull;
import io.reactivex.schedulers.Schedulers;

/**
 * @author woong
 */
public class RxSchedulers {

    final static ObservableTransformer schedulersTransformer = new ObservableTransformer() {
        @Override
        public ObservableSource apply(@NonNull Observable upstream) {
            return upstream.subscribeOn(Schedulers.newThread())
                    .observeOn(AndroidSchedulers.mainThread());

        }
    };

    static <T> ObservableTransformer<T, T> applySchedulers() {
        return schedulersTransformer;
    }

    @SuppressWarnings("unchecked")
    public static <T> ObservableTransformer<T, T> io_main() {

        return (ObservableTransformer<T, T>) applySchedulers();
    }

}

File: app/src/main/java/cn/com/woong/readhub/network/RxSchedulers.kt
package cn.com.woong.readhub.network

import io.reactivex.ObservableTransformer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

object RxSchedulers {
    fun <T> applySchedulers(): ObservableTransformer<T, T> {
        return ObservableTransformer { upstream -> upstream.subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread()) }
    }

    fun <T> io_main(): ObservableTransformer<T, T> {
        return applySchedulers<Any>() as ObservableTransformer<T, T>
    }
}